src/QMCDrivers/QMCDriverNew.cpp

```diff
@@ -238,9 +238,8 @@ void QMCDriverNew::recordBlock(int block)
     population_.saveWalkerConfigurations(walker_configs_ref_);
     setWalkerOffsets(walker_configs_ref_, myComm);
     wOut->dump(walker_configs_ref_, block);
-#ifndef USE_FAKE_RNG
-    RandomNumberControl::write(getRngRefs(), get_root_name(), myComm);
-#endif
+    if (qmcdriver_input_.get_dump_config())
+      RandomNumberControl::write(getRngRefs(), get_root_name(), myComm);
```

Comment:
I'm less confident about this change.

Reply:
RandomNumberControl::write assumes that any random ref it writes has the same state size and other characteristics as `random_global` which seems like poor form since I think everywhere its done the actual type being operated on is available.  But it seems to me this must just protect for the compile time lack of a save method in FakeRandom.  I can't see where there is actually unit test coverage of this.  If QMCDriverNew had better unit test coverage and it used FakeRandom I think the solution would be to just give FakeRandom a mock save function that did something to the vector argument of that method.